### ISSUE TITLE
Cannot use ScalarTransform4f in dicts to instantiate rectangle plugins with double precision variants

### ISSUE THREAD

**Issue by leroyvn** [mitsuba-renderer/mitsuba2#152]
I cannot use a `ScalarTransform4f in dicts to instantiate shape plugins when using double-precision variants.

# System configuration

Platform: macOS 10.15.4
Compiler: Apple clang version 11.0.0 (clang-1100.0.33.17)
Python version: Python 3.7.6 :: Anaconda, Inc.
Mitsuba 2 version: e345a73d32d631a57f3bebcb7ec8fc4c3bf41995
Compiled variants: 
* `scalar_mono`
* `scalar_mono_double`
* `scalar_mono_polarized`
* `scalar_mono_polarized_double`
* `scalar_rgb`
* `scalar_rgb_double`
* `scalar_spectral`
* `scalar_spectral_double`
* `packet_rgb`
* `packet_spectral`

# Problem

I cannot use `ScalarTransform4f` to set the `to_world` parameter of a `rectangle` plugin when using the following variants:
- `scalar_mono_double`
- `scalar_mono_polarized_double`
- `scalar_rgb_double`
- `scalar_spectral_double`

# Steps to reproduce

1. Compile Mitsuba with the aforementioned variants.
2. Run the following script:

```python
import mitsuba

results = []

for variant in mitsuba.variants():
    mitsuba.set_variant(variant)
    from mitsuba.core import ScalarTransform4f

    try:
        mitsuba.core.xml.load_dict({
            "type": "rectangle", 
            "to_world": ScalarTransform4f.scale(0.5)
        })
    except RuntimeError:
        results.append(f"x {variant}")
        continue
    
    results.append(f"o {variant}")

print("\n".join(sorted(results)))
```

The expected output is as follows:
```
2020-05-26 13:42:18 INFO  main  [PluginManager] Loading plugin "plugins/rectangle.dylib" ..
2020-05-26 13:42:18 INFO  main  [PluginManager] Loading plugin "plugins/diffuse.dylib" ..
2020-05-26 13:42:19 INFO  main  [PluginManager] Loading plugin "plugins/uniform.dylib" ..
o packet_rgb
o packet_spectral
o scalar_mono
o scalar_mono_polarized
o scalar_rgb
o scalar_spectral
x scalar_mono_double
x scalar_mono_polarized_double
x scalar_rgb_double
x scalar_spectral_double
```

Additional check: run the command
```
python -c "import mitsuba; mitsuba.set_variant('scalar_mono_double'); mitsuba.core.xml.load_dict({'type': 'rectangle', 'to_world': mitsuba.core.ScalarTransform4f.scale(0.5)})"
```

Expected output:
```
2020-05-26 13:45:30 INFO  main  [PluginManager] Loading plugin "plugins/rectangle.dylib" ..
Traceback (most recent call last):
  File "<string>", line 1, in <module>
RuntimeError: ​[xml_v.cpp:259] Unkown value type: <class 'mitsuba.core.Transform4f'>
```

**Comment by Speierers** [comment]
Hi @leroyvn ,

Thanks for reporting this. It should be fixed now: (d5b154d4)

**Comment by leroyvn** [comment]
Thanks!
